lmeval-evaluation-job.adoc

```diff
@@ -0,0 +1,185 @@
+* `metrics` (optional):  Specify a Unitxt task from the link:https://www.unitxt.ai/en/latest/catalog/catalog.__dir__.html[Unitxt catalog]. Use the metric's ID as the value. A Unitxt task has a set of pre-defined metrics. Only specify a set of metrics if you need different metrics.
+* `format` (optional): Specify a Unitxt format from the link:https://www.unitxt.ai/en/latest/catalog/catalog.__dir__.html[Unitxt catalog]. Use the format's ID as the value.
+* `loaderLimit` (optional): Specifies the maximum number of instances per stream to be returned from the loader (used to reduce loading time in large datasets).
+* `numDemos` (optional): Number of fewshot to be used.
+* `demosPoolSize` (optional): Size of the fewshot pool.
+
+
+| `numFewShot`
+| Sets the number of few-shot examples to place in context. If you are using a task from Unitxt, don't use this field. Use `numDemos` under the `taskRecipes` instead.
+
+| `limit`
+| Instead of running the whole dataset, set a limit to run the tasks. Accepts an integer, or a float between 0.0 and 1.0.
+
+| `genArgs`
+| Map to `--gen_kwargs` parameter for the lm-evaluation-harness. Read more in the link:https://github.com/EleutherAI/lm-evaluation-harness/blob/main/docs/interface.md#command-line-interface[lm-evaluation-harness documentation].
+
+| `logSamples`
+| If this flag is passed, then the model's outputs, and the text fed into the model, will be saved at per-document granularity.
+
+| `batchSize`
+| Batch size for the evaluation. The auto:`N` batch size is not used for API models, but numeric batch sizes are used for APIs. Only int batch size supported at the moment.
+
+| `pod`
+| Specify extra information for the lm-eval job's pod. 
+* `container`: Extra container settings for the lm-eval container.
+** `env`: Specify environment variables. It uses the `EnvVar` data structure of kubernetes.
+** `volumeMounts`: Mount the volumes into the lm-eval container
+** `resources`: Specify the resources for the lm-eval container.
+* `volumes`: Specify the volume information for the lm-eval and other containers. It uses the `Volume`  data structure of kubernetes.
+* `sideCars`: A list of containers that run along with the lm-eval container. It uses the `Container` data structure of kubernetes.
+
+
+| `outputs`
+| This section defines custom output locations for the evaluation results storage. At the moment only Persistent Volume Claims (PVC) are supported.
+
+| `outputs.pvcManaged`
+| Create an operator-managed PVC to store this job's results. The PVC will be named `<job-name>-pvc` and will be owned by the `LMEvalJob`. After job completion, the PVC will still be available, but it will be deleted upon deleting the `LMEvalJob`. Supports the following fields:
```

Comment:
```suggestion
| Creates an operator-managed PVC to store this job's results. The PVC is named `<job-name>-pvc` and is owned by the `LMEvalJob`. After the job finishes, the PVC is still be available, but it is deleted with the `LMEvalJob`. Supports the following fields:
```
Can we find a different wording than "is owned by" - in addition to being passive voice, it anthropomorphizes the job. More info: https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism